Symfony/CS/Fixer/Contrib/ArrayElementSpaceFixer.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the PHP CS utility.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Fixer\Contrib;
+
+use Symfony\CS\AbstractFixer;
+use Symfony\CS\Tokenizer\Token;
+use Symfony\CS\Tokenizer\Tokens;
+
+/**
+ * Fixer for rules defined in PSR2 ¶4.6.
+ *
+ * @author Adam Marczuk <adam@marczuk.info>
+ */
+class ArrayElementSpaceFixer extends AbstractFixer
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function fix(\SplFileInfo $file, $content)
+    {
+        $tokens = Tokens::fromCode($content);
+
+        for ($index = $tokens->count() - 1; $index >= 0; --$index) {
+            if (!$tokens->isArray($index)) {
+                continue;
+            }
+
+            $this->fixSpacing($index, $tokens);
+        }
+
+        return $tokens->generateCode();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDescription()
+    {
+        return 'In array declaration, there MUST NOT be a space before each comma, there MUST be one space after each comma and there should be space around double arrow.';
+    }
+
+    /**
+     * Method to fix spacing in array declaration.
+     *
+     * @param int    $index
+     * @param Tokens $tokens
+     */
+    private function fixSpacing($index, Tokens $tokens)
+    {
+        if ($tokens->isShortArray($index)) {
+            $startIndex = $index;
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_SQUARE_BRACE, $startIndex);
+        } else {
+            $startIndex = $index + 1;
+            $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startIndex);
+        }
+
+        if ($tokens[$startIndex + 1]->isWhitespace()) {
```

Comment:
I've left it there because this is about fixing array's start and end token not it's elements. Code below is looping over array's elements backwards but it's separate logic.